pkg/preparer/setup.go

```diff
@@ -26,16 +26,30 @@ type LogDestination struct {
 }
 
 type PreparerConfig struct {
-	NodeName             string           `yaml:"node_name"`
-	ConsulAddress        string           `yaml:"consul_address"`
-	ConsulTokenPath      string           `yaml:"consul_token_path,omitempty"`
-	HooksDirectory       string           `yaml:"hooks_directory"`
-	KeyringPath          string           `yaml:"keyring,omitempty"`
-	CAPath               string           `yaml:"ca_path,omitempty"`
-	PodRoot              string           `yaml:"pod_root,omitempty"`
-	AuthorizedDeployers  []string         `yaml:"authorized_deployers,omitempty"`
-	ForbiddenPodIds      []string         `yaml:"forbidden_pod_ids,omitempty"`
-	ExtraLogDestinations []LogDestination `yaml:"extra_log_destinations,omitempty"`
+	NodeName             string                 `yaml:"node_name"`
+	ConsulAddress        string                 `yaml:"consul_address"`
+	ConsulTokenPath      string                 `yaml:"consul_token_path,omitempty"`
+	HooksDirectory       string                 `yaml:"hooks_directory"`
+	CAPath               string                 `yaml:"ca_path,omitempty"`
+	PodRoot              string                 `yaml:"pod_root,omitempty"`
+	Auth                 map[string]interface{} `yaml:"auth,omitempty"`
```

Comment:
Changing it from `interface{}` to `string` wouldn't impact the current use cases.